### ISSUE TITLE
TB78 : HEADER tag do not work anymore

### ISSUE THREAD

**Issue by userQAY** [thundernest/quicktext#169]
tag HEADER forks fine with **TO**  [[HEADER=to|abc@abc.com]]

but 
with **CC** [[HEADER=cc|abc@abc.com]]
it writes to header **two** mail addresses, one for TO and one for CC

with **BCC** [[HEADER=bcc|abc@abc.com]]
it writes to header **three**  mail addresses, one for TO , one for CC and one for BCC

could you fix it please?

**Comment by jobisoft** [comment]
thats odd

**Comment by SamuelPlentz** [comment]
Could reproduce at my work pc, couldn't reproduce at home.

**Comment by SamuelPlentz** [comment]
Work PC: I also could restart Thunderbird and it could still remember the additional values of TO, CC and BCC.

**Comment by SamuelPlentz** [comment]
Work PC: Between restarts the data is written to this folder: "Thunderbird Profile\quicktext\templates.xml"

Mention there is an additional "headers" tag, containing the cc-address while the template itself only contains a to-address.
```
<?xml version="1.0"?>
<quicktext version="2">
	<filetype>templates</filetype>
	<menu>
		<title><![CDATA[Test]]></title>
		<texts>
			<text shortcut="" type="1">
				<name><![CDATA[Test]]></name>
				<body><![CDATA[...[[HEADER=to|abc@abc.com]]...]]></body>
				<headers>
					<header>
						<type><![CDATA[cc]]></type>
						<value><![CDATA[abc@abc.com]]></value>
					</header>
				</headers>
			</text>
			<text shortcut="" type="1">
				<name><![CDATA[Test2]]></name>
				<body><![CDATA[...]]></body>
			</text>
		</texts>
	</menu>
</quicktext>
```

**Comment by jobisoft** [comment]
I am afraid its a corruption in the template files. Can you create a new template and move (copy paste in the quicktext window) the template content from the old to the new template? That fixed it for me. I could not reproduce the error in the new template.

Can you confirm?

**Comment by SamuelPlentz** [comment]
The bug results in corrupted template files and you can fix it by the described workaround. Confirmed.

But there seems to be a reason that header informations appear in the template file. The following code (line 777 and following) looks to be written for exactly this purpose. (there is similar code in the `parseImport` function)
https://github.com/jobisoft/quicktext/blob/4af3d7188cd6fb8dfb0b74b97392ed06133590c2/chrome/content/modules/wzQuicktext.jsm#L763-L790

I don't understand why caching of header informations is needed/implemented. IMO the key questions are why is caching needed and what triggers the caching. (as I can only partly reproduce the problem)

**Comment by SamuelPlentz** [comment]
> tag HEADER forks fine with **TO** [[HEADER=to|[abc@abc.com](mailto:abc@abc.com)]]
> 
> but
> with **CC** [[HEADER=cc|[abc@abc.com](mailto:abc@abc.com)]]
> it writes to header **two** mail addresses, one for TO and one for CC
> 
> with **BCC** [[HEADER=bcc|[abc@abc.com](mailto:abc@abc.com)]]
> it writes to header **three** mail addresses, one for TO , one for CC and one for BCC

With #185 it should insert only 1 address in each of this cases.

**Comment by jobisoft** [comment]
Merged #185.

**Comment by SamuelPlentz** [comment]
Could be closed now.
